rust/src/qfunction.rs

```diff
@@ -168,25 +170,25 @@ impl fmt::Display for QFunctionByName {
 // -----------------------------------------------------------------------------
 impl QFunctionCore {
     // Common implementation
-    pub fn apply(&self, Q: i32, u: &Vec<crate::vector::Vector>, v: &Vec<crate::vector::Vector>) {
+    pub fn apply(&self, Q: i32, u: &[Vector], v: &[Vector]) {
+        let mut u_c = [std::ptr::null_mut(); 16];
```

Comment:
I specifically used `Vec` as the input/output of QFunctions so that the user would not need to know the magic number 16.

Reply:
Woops, wrong place